script/BaseScript.s.sol

```diff
@@ -374,60 +386,59 @@ contract BaseScript is BaseData, Utils {
         internal
-        pure
+        view
         returns (bytes32 _salt)
     {
         _salt = createSalt(_deployerAddress, "L1YnOFTAdapterProxy");
     }
 
-    function createL1YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL1YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L1YnOFTAdapter");
     }
 
     function createL2YnOFTAdapterUpgradeableProxySalt(address _deployerAddress)
         internal
-        pure
+        view
         returns (bytes32 _salt)
     {
         _salt = createSalt(_deployerAddress, "L2YnOFTAdapterProxy");
     }
 
-    function createL2YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnOFTAdapterUpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnOFTAdapter");
     }
 
-    function createL2YnERC20UpgradeableProxySalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnERC20UpgradeableProxySalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnERC20Proxy");
     }
 
-    function createL2YnERC20UpgradeableSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnERC20UpgradeableSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnERC20");
     }
 
-    function createL1YnOFTAdapterTimelockSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL1YnOFTAdapterTimelockSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L1YnOFTTimelock");
     }
 
-    function createL2YnOFTAdapterTimelockSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
+    function createL2YnOFTAdapterTimelockSalt(address _deployerAddress) internal view returns (bytes32 _salt) {
         _salt = createSalt(_deployerAddress, "L2YnOFTTimelock");
     }
 
-    // function createL2YnERC20TimelockSalt(address _deployerAddress) internal pure returns (bytes32 _salt) {
-    //     _salt = createSalt(_deployerAddress, "L2YnERC20Timelock");
-    // }
+    function createSalt(address _deployerAddress, string memory _label) internal view returns (bytes32 _salt) {
+        require(bytes(baseInput.erc20Symbol).length > 0, "Invalid ERC20 Symbol");
 
-    function createSalt(address _deployerAddress, string memory _label) internal pure returns (bytes32 _salt) {
         _salt = bytes32(
             abi.encodePacked(
-                bytes20(_deployerAddress), bytes12(bytes32(keccak256(abi.encode(_label, LABEL, VERSION))))
+                bytes20(_deployerAddress),
```

Comment:
why was this changed ?

not a clear reason here

Reply:
i can't tell if there's a reason for this change, if there's no particular reason, let's not make it

Reply:
The reason for this change is that if the version and label were kept the same, the salt is the same for deployments for different tokens, which created conflicts while deploying the contracts. 

Therefore I made the salt dependant on the symbol of the token being deployed.